### PR TITLE
Issue 21159 forms csp security unsafe eval is required when using dotcms forms

### LINKED ISSUE #21159: Forms CSP/Security: 'unsafe-eval' is required when using dotcms forms
**Describe the bug**

When using dotcms forms on pages, this uses eval() on the successCallback JS string when it's time to execute. This will be prevented from executing if `'unsafe-eval'` is not enabled for sites using CSP..

**To Reproduce**
1. Create a form contentlet, doing whatever for success callback
2. Add form to a page
3. Submit form
4. Error displays

**Expected behavior**
`eval()` shouldn't be needed. Executing the success callback without `eval()` would be ideal.

**Screenshots**

image

Context: https://github.com/dotCMS/core-web/blob/2e0b6195de6ee470833722ccdb074c9dfac1fd3c/libs/dotcms-webcomponents/src/components/contenttypes-fields/dot-form/dot-form.tsx#L179

Edit: Being CSP conscious, the script tags being injected with dotcms forms should also come with a hash. As a workaround, I had to take the code generated by `content_form.vtl` so I could inject a generated nonce we are using.

image

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/form/FormResource.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.glassfish.jersey.server.JSONP;
+
+@Path("/v1/form")
+public class FormResource {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/form/FormResource.java:35]: Doc

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/form/FormResource.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    public FormResource () {
+        this.webResource = new WebResource();
+    }
+
+    @GET
+    @Path("/{idOrVar}/successCallback")
+    @NoCache
+    @Produces({"application/javascript"})
+    public final Response createType(@Context final HttpServletRequest req,
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/form/FormResource.java:48]: doc

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/form/FormResource.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        final ContentType contentType = APILocator.getContentTypeAPI(initDataObject.getUser())
+                .find(idOrVar);
+
+        if (BaseContentType.FORM  != contentType.baseType()){
+            throw new NotFoundInDbException("The form not exists: " + idOrVar);
+        }
+
+        final List<Field> fields = contentType.fields().stream()
+                .filter(field -> FORM_SUCCESS_CALLBACK.equals(field.variable()))
+                .limit(1)
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/form/FormResource.java:69]: isn't easier just findFirst? such as
```
final Optional<Field> fields = contentType.fields().stream()
                .filter(field -> FORM_SUCCESS_CALLBACK.equals(field.variable()))
                .findFirst();

if (fields.notPresent()) {

    throw ....
    ```

**Review by jdotcms — COMMENTED** [review]
I have left some comments

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/form/FormResource.java (hunk 1 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/form/FormResource.java:39]: done https://github.com/dotCMS/core/pull/21277/commits/88b33e094f38a21d16b1344a52fac3fa4c504b01#diff-1115868267dc10d76e4a8503d99607c6d03ab9b3c629c32d6f492e7248427f25R35

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/form/FormResource.java (hunk 3 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/form/FormResource.java]: done https://github.com/dotCMS/core/pull/21277/commits/88b33e094f38a21d16b1344a52fac3fa4c504b01#diff-1115868267dc10d76e4a8503d99607c6d03ab9b3c629c32d6f492e7248427f25R48

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/form/FormResource.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/form/FormResource.java:63]: done https://github.com/dotCMS/core/pull/21277/commits/88b33e094f38a21d16b1344a52fac3fa4c504b01#diff-1115868267dc10d76e4a8503d99607c6d03ab9b3c629c32d6f492e7248427f25R48

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/form/FormResource.java (hunk 3 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/form/FormResource.java]: done https://github.com/dotCMS/core/pull/21277/commits/88b33e094f38a21d16b1344a52fac3fa4c504b01#diff-1115868267dc10d76e4a8503d99607c6d03ab9b3c629c32d6f492e7248427f25R75

**Review by jdotcms — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Pull request merged**